CONTRIBUTING.rst

```diff
@@ -806,6 +806,45 @@ If this function is designed to be called by "end-users" (i.e. DAG authors) then
       ...
       # You SHOULD not commit the session here. The wrapper will take care of commit()/rollback() if exception
 
+Don't use time() for duration calcuations
+-----------------------------------------
+
+If you wish to compute the time difference between two events with in the same process, use
+``time.monotonic()``, not ``time.time()`` nor ``timzeone.utcnow()``.
+
+If you are measuring duration for performance reasons, then ``time.perf_counter()`` should be used. (On many
+platforms, this uses the same underlying clock mechanism as monotonic, but ``perf_counter`` is guaranteed to be
+the highest accuracy clock on the system, monotonic is simply "guaranteed" to not go backwards.)
+
+If you wish to time how long a block of code takes, use ``Stats.timer()`` -- either with a metric name, which
+will be timed and submitted automatically:
+
+.. code-block:: python
+
+    from airflow.stats import Stats
+
+    ...
+
+    with Stats.timer("my_timer_metric"):
+        ...
+
+or without, to access the ``duration`` property:
+
+.. code-block:: python
+
+    from airflow.stats import Stats
+
+    ...
+
+    with Stats.timer("my_timer_metric") as timer:
```

Comment:
```suggestion
or without, to access the ``duration`` property:

.. code-block:: python

    from airflow.stats import Stats

    ...

    with Stats.timer() as timer:
```

To comply with "or without..."?

Reply:
Hmmm yeah, I'll tidy this up. I've got duplication between here and in airflow.stats -- I might just link from here?

Reply:
Left these two common cases here (with a tidy up as suggested), and linked to airflow/stats.py.